net/query.go

```diff
@@ -56,17 +57,23 @@ func (h *Host) Query(who peer.ID) (*QueryResponse, error) {
 }
 
 func receiveQueryResponse(stream libp2pnetwork.Stream) (*QueryResponse, error) {
-	msg, err := readStreamMessage(stream, maxMessageSize)
-	if err != nil {
-		return nil, fmt.Errorf("error reading QueryResponse: %w", err)
-	}
-
-	resp, ok := msg.(*QueryResponse)
-	if !ok {
-		return nil, fmt.Errorf("expected %s message but received %s",
-			message.TypeToString(message.QueryResponseType),
-			message.TypeToString(msg.Type()))
+	const queryResponseTimeout = time.Second * 15
```

Comment:
We have a `queryTimeout` at the package level that appears to be only for the connect. That should probably be renamed. I'd either put both of them together at the package level, or move the connect timeout to the function level.

Reply:
updated to use one `connectionTimeout` var